From: Tom Herbert <tom@herbertland.com>
Date: Mon, 14 Dec 2015 11:19:40 -0800
Subject: net: Add skb_inner_transport_offset function
Patch-mainline: v4.5-rc1
Git-commit: 55dc5a9f2f2afd32d7b1bda44a5fc95e67a3371f
References: bsc#966170 FATE#320225 bsc#966172 FATE#320226

Same thing as skb_transport_offset but returns the offset of the inner
transport header (when skb->encpasulation is set).

Signed-off-by: Tom Herbert <tom@herbertland.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 include/linux/skbuff.h |    5 +++++
 1 file changed, 5 insertions(+)

--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -1964,6 +1964,11 @@ static inline unsigned char *skb_inner_t
 	return skb->head + skb->inner_transport_header;
 }
 
+static inline int skb_inner_transport_offset(const struct sk_buff *skb)
+{
+	return skb_inner_transport_header(skb) - skb->data;
+}
+
 static inline void skb_reset_inner_transport_header(struct sk_buff *skb)
 {
 	skb->inner_transport_header = skb->data - skb->head;
